src/WatchDogs.Infrastructure.DxTrade/DxTradeAuthenticator.cs

```diff
@@ -0,0 +1,86 @@
+﻿using Contracts;
+using Microsoft.Extensions.Options;
+using System.Net.Http.Headers;
+using System.Security.Authentication;
+using System.Text;
+using System.Text.Json;
+
+namespace Infrastructure.DxTrade
+{
+    public class DxTradeAuthenticator : IDxTradeAuthenticator
+    {
+        private const string CookieGroupName = "SET-COOKIE";
+        private const string SessionTokenHeaderName = "JSESSIONID";
+
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly DxTradeConnectionOptions _connectionOptions;
+        private InMemorySessionTokenStorage _sessionToken;
+
+
+        public DxTradeAuthenticator(IOptions<DxTradeConnectionOptions> configuration, IHttpClientFactory httpClientFactory)
+        {
+            _connectionOptions = configuration.Value;
+            _httpClientFactory = httpClientFactory;
+            _sessionToken = new InMemorySessionTokenStorage();
+        }
+
+        public async Task AuthenticateAsync()
+        {
+            //Due to the inconsistencies in documentation, specifically base URL, request body and response body,
+            //authentication done with the help of: https://github.com/zLeki/DXTrade-Python-Demo/blob/main/main.py
+
+            try
+            {
+                using var client = _httpClientFactory.CreateClient("DxTradeAuthenticationClient");
+
+                string jsonData = JsonSerializer.Serialize(_connectionOptions);
+
+                var stringContent = new StringContent(jsonData, Encoding.UTF8, "application/json");
+
+                var response = await client.PostAsync("login", stringContent);
+
+                if (response.IsSuccessStatusCode)
+                {
+                    //string sessionToken = GetSessionToken(response.Headers);
+
+                    var sessionToken = new InMemorySessionTokenStorage();
+
+                    await sessionToken.SetSessionTokenAsync(GetSessionToken(response.Headers));
+
+                    _sessionToken = sessionToken;
+
+                }
+
+                else
+                {
+                    throw new AuthenticationException(response.StatusCode.ToString());
```

Comment:
Btw you can call `response.EnsureSuccessStatusCode();` that will throw `HttpRequestException` if it is not successful. Since we have it wrapped in try-catch block and you will also add catch for `Excetpion`, you could call this method instead of if-else branching.

Reply:
Done.